File: Backend/Events/Converters/PartyConverter.cs
using Backend.Domain.Models;
using Backend.Events.Converters.Parties;
using Backend.Events.DTO;

namespace Backend.Events.Converters;

public static class PartyConverter
{
    public static PartyDTO ToDTO(Party party) => new()
    {
        Slots = party.Slots.Select(DigimonSlotConverter.ToDTO).ToList()
    };
}

File: Frontend/src/events/converters/party.converter.ts
import type { DeepRequired } from "@/events/dto/deep-required";
import type { PartyDTO } from "@/events/dto/party.dto";
import type { Party } from "@/models";
import { DigimonSlotConverter } from "./parties/digimon-slot.converter";

export class PartyConverter {
  public static convert(partyDto: DeepRequired<PartyDTO>): Party {
    return {
      slots: partyDto.slots.map((digimonSlotDto) => DigimonSlotConverter.convert(digimonSlotDto)),
    };
  }
}
